drivers/gpio/gpio_gd32.c

```diff
@@ -55,6 +57,7 @@ struct gpio_gd32_data {
 	sys_slist_t callbacks;
 };
 
+#ifdef CONFIG_GD32_EXTI
```

Comment:
`#ifdef` can be removed

Reply:
This `#ifdef` suppressing `unused-function` warning.
It should be better not to remove it.